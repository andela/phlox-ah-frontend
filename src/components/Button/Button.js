import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

export const Button = props => (
  <div className="row">
    <button
      onClick={props.onClick}
      className={`btn waves-effect waves-light form-button ${props.buttonClassName || ''}`}
      type={props.type || 'button'}>
      {props.name}
    </button>
  </div>
);


Button.propTypes = {
  buttonClassName: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
  name: PropTypes.any
};
