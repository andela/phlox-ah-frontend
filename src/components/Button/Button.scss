$primary-color: #06576D;

div.row {

  button.form-button {
    display: block;
    width: 89%;
    border-radius: 20px;
    margin: auto;
    background-color: $primary-color;
    text-transform: capitalize;
    font-weight: 600;
  }

}
