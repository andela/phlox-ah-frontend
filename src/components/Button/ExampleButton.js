import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line
const Button = props => {
  return (
  <div>
    <button>{ props.text }</button>
  </div>
  );
};

Button.propTypes = {
  text: PropTypes.string,
};

export default Button;
