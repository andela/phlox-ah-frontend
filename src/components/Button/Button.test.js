import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import { Button } from './Button';

const props = {
  onClick: jest.fn(),
  buttonClassName: 'class-name',
  type: 'button',
  name: 'button'
};

let component;


describe('<Button/>', () => {
  beforeEach(() => {
    component = shallow(
        <Button {...props} />
    );
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have a div tag with class row', () => {
    expect(component.find('div.row').exists()).toBe(true);
  });
  it('should have a button tag', () => {
    expect(component.find('button').exists()).toBe(true);
  });
});
