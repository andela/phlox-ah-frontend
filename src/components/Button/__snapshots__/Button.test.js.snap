// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render without throwing an error 1`] = `
<div
  className="row"
>
  <button
    className="btn waves-effect waves-light form-button class-name"
    onClick={[MockFunction]}
    type="button"
  >
    button
  </button>
</div>
`;
